graphene/types/schema.py

```diff
@@ -391,6 +411,23 @@ def resolve_type(self, resolve_type_func, type_name, root, info, _type):
         return_type = self[type_name]
         return default_type_resolver(root, info, return_type)
 
+    def add_prefix_to_type_name(self, name):
+        if self.type_name_prefix:
+            return self.type_name_prefix[0].upper() + self.type_name_prefix[1:] + name
+        return name
+
+    def add_prefix_to_field_name(self, name):
+        if self.type_name_prefix:
+            if self.auto_camelcase:
+                return self.get_name(
+                    self.type_name_prefix[0].lower()
+                    + self.type_name_prefix[1:]
+                    + "_"
+                    + name
+                )
+            return self.get_name(self.type_name_prefix + name)
+        return self.get_name(name)
```

Comment:
I don't think there is a need to deal with the `auto_camelcase` case here as it is already dealt with in the `get_name`.

```suggestion
    def add_prefix_to_field_name(self, name):
        if self.type_name_prefix:
           return self.get_name(
                self.type_name_prefix[0].lower()
                + self.type_name_prefix[1:]
                + "_"
                + name
            )
        return self.get_name(name)
```

also note that in your current implementation where you don't `lower` the first character in the case we are not in auto camel case means the output is not exactly what you give in the doc:
```
    type Query {
        myPrefixInner: MyPrefixMyType
    }
```
indeed that would be `MyPrefixInner`...

Reply:
> indeed that would be MyPrefixInner...

By default `auto_camelcase=True` so the query name would be `myPrefixInner` indeed isn't it?

Reply:
yes but my point was if `auto_camelcase` is:
- `True` ->  `myPrefixInner`
- `False`  -> `MyPrefixInner`
which doesn't seem very intuitive... no?

Reply:
I'm confused, what would be more intuitive in your view? 

Because 

* `auto_camelcase=True` -> `myPrefixInner` (`camelCase`)
* `auto_camelcase=False` -> we don't modify the prefix -> `MyPrefixInner` (`PascalCase`) 

Feels the most natural to me 🤔

Reply:
Gentle ping on that one @tcleonard